_posts/2021-01-18-getting-started-kafka.adoc

```diff
@@ -0,0 +1,303 @@
+Create the `org.acme.MovieProducer` class with the following content:
+
+[source, java]
+----
+package org.acme;
+
+import io.smallrye.reactive.messaging.kafka.Record;
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+@ApplicationScoped
+public class MovieProducer {
+
+    @Inject @Channel("movies-out")
+    Emitter<Record<Integer, String>> emitter;
+
+    public void sendMovieToKafka(Movie movie) {
+        emitter.send(Record.of(movie.year, movie.title));
+    }
+}
+----
+
+In this class, we inject an `Emitter`, i.e., an object responsible for sending a message to a channel.
+This emitter is attached to the `movies-out` channel (and so will send messages to Kafka).
+We are sending `Record` objects containing the movie's publication year as key and its title as value.
+
+So, the rest of our application can simply use the `sendMovieToKafka` method to send a movie to our Kafka topic.
+
+== Step 5 - Consuming movies
+
+Let's now look at the other side and retrieve the movies from Kafka.
+
+[source, java]
+----
+package org.acme;
+
+import io.smallrye.reactive.messaging.kafka.Record;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.jboss.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class MovieConsumer {
+
+    private final Logger logger = Logger.getLogger(MovieConsumer.class);
+
+    @Incoming("movies-in")
+    public void receive(Record<Integer, String> record) {
+        logger.infof("Got a movie: %d - %s", record.key(), record.value());
+    }
+}
+----
+
+Here, we use the `@Incoming` annotation to indicate to Quarkus to call the `receive` method for every received record.
+
+== Step 6 - Sending movies from a REST endpoint
+
+It's quite common to send messages to Kafka from a REST endpoint.
+To do this, create the `org.acme.MovieResource` class with the following content:
+
+[source, java]
+----
+package org.acme;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("/")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+public class MovieResource {
+
+    @Inject
+    MovieProducer producer;
+
+    @POST
+    public Response send(Movie movie) {
+        producer.sendMovieToKafka(movie);
+        // Return an 202 - Accepted response.
+        return Response.accepted().build();
+    }
+}
+----
+
+== Step 7 - Let's get this running!
+
+Well, first, we need a Kafka broker.
+You can follow the https://kafka.apache.org/quickstart[Apache Kafka quickstart], or use the following `docker-compose.yaml` file:
+
+[source, yaml]
+----
+version: '2'
+
+services:
+
+ zookeeper:
+ image: strimzi/kafka:0.19.0-kafka-2.5.0
```

Comment:
Should this be updated to "0.20.1-kafka-2.6.0"?

Reply:
Yes, updated.